### ISSUE TITLE
Bug: no-unpublished-bin / no-unpublished-require - false positive with brace expansion and extglob in files field

### ISSUE THREAD

**Issue by sbonasu** [eslint-community/eslint-plugin-n#555]
### Environment

Node version: v22.22.3
npm version: 10.9.8
ESLint version: 8.57.1
eslint-plugin-n version: 15.7.0
Operating System: macOS 26.5.2

### What rule do you want to report?

no-unpublished-bin / no-unpublished-require

### Link to Minimal Reproducible Example

https://github.com/sbonasu/eslint-plugin-n-brace-repro

## Description

The `node/no-unpublished-bin` (and likely `node/no-unpublished-require`) rule produces false positives when the `files` field in `package.json` uses npm-supported brace expansion or extglob patterns such as `{bin,src}/**/!(*.test).js`.

## Minimal Reproduction

**package.json:**
```json
{
  "name": "repro",
  "version": "1.0.0",
  "files": ["{bin,src}/**/!(*.test).js"],
  "bin": {
    "foo": "bin/foo.js"
  }
}
```

**bin/foo.js:**
```js
#!/usr/bin/env node
console.log("hello");
```

**.eslintrc.js:**
```js
module.exports = { rules: { "n/no-unpublished-bin": "error" } };
```

**Result:**
```
$ npx eslint bin/foo.js
error  npm ignores 'bin/foo.js'  n/no-unpublished-bin
```

**Expected:** No error, because `npm pack --dry-run` confirms `bin/foo.js` IS included in the tarball.

## Root Cause

`eslint-plugin-n`'s [`lib/util/get-npmignore.js` `parseWhiteList()` function](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/util/get-npmignore.js#L82-L108) uses the [`ignore`](https://www.npmjs.com/package/ignore) package (gitignore-style matching) to evaluate the `files` field. However, the `ignore` package does **not** support:

1. **Brace expansion** (`{bin,src}`) — not a gitignore feature
2. **Extglob negation** (`!(*.test)`) — not a gitignore feature

npm itself uses [`minimatch`](https://www.npmjs.com/package/minimatch) which fully supports both of these patterns. This mismatch means npm correctly includes files, but `eslint-plugin-n` incorrectly thinks they are excluded/ignored.

### Verification

| Pattern: `{bin,src}/**/!(*.test).js` | `minimatch` (npm behavior) | `ignore` pkg (eslint-plugin-n behavior) |
|------|-------------------|-------------------------------|
| `bin/foo.js` | ✅ matches (included) | ❌ not matched (treated as ignored) |
| `bin/foo.test.js` | ❌ not matched (excluded) | ❌ not matched |

Even simple brace expansion fails — the `ignore` package cannot match `bin/foo.js` against `/{bin,src}/**`.

## Suggested Fix

Replace the use of the `ignore` package with `minimatch` (or `picomatch`) when evaluating the `files` field, since npm uses minimatch-compatible glob matching for `files` — not gitignore-style matching. The `ignore` package is still appropriate for `.npmignore` evaluation, but not for the `files` array in `package.json`.

Alternatively, expand brace/extglob patterns (e.g., via [`micromatch.braces()`](https://github.com/micromatch/micromatch)) before passing them to the `ignore` instance.

### Participation

- [ ] I am willing to submit a pull request for this issue.
